components/rhel/misc/rhel-k8s-singlenode-setup/Vagrantfile

```diff
@@ -41,8 +37,23 @@ Vagrant.configure(2) do |config|
     # Enable use of more than one virtual CPU in a virtual machine.
     vbox.customize ["modifyvm", :id, "--ioapic", "on"]
   end
+
   config.vm.network "private_network", type: "dhcp"
 
+  if ENV.has_key?('SUB_USERNAME') && ENV.has_key?('SUB_PASSWORD')
+    config.registration.username = ENV['SUB_USERNAME']
+    config.registration.password = ENV['SUB_PASSWORD']
+  end
+
+  # Proxy Information from environment
+  if ENV.key?('PROXY')
+    config.registration.proxy = PROXY = ENV['PROXY']
+    config.registration.proxyUser = PROXY_USER = ENV['PROXY_USER']
+    config.registration.proxyPassword = PROXY_PASSWORD = ENV['PROXY_PASSWORD']
+  else
+    PROXY = PROXY_USER = PROXY_PASSWORD = ''
+  end
+
```

Comment:
What if we only have proxy server without authentication so how this will work out. does it set `PROXY_USERNAME` and `PROXY_PASSWORD` as `''`?